yarn-project/simulator/src/avm/apps_tests/token.test.ts

```diff
@@ -0,0 +1,84 @@
+import { AztecAddress, type ContractInstanceWithAddress, Fr } from '@aztec/circuits.js';
+import { TokenContractArtifact } from '@aztec/noir-contracts.js/Token';
+
+import { AvmSimulationTester } from '../fixtures/avm_simulation_tester.js';
+
+describe('AVM simulator apps tests: TokenContract', () => {
+  const admin = AztecAddress.fromNumber(42);
+  const sender = AztecAddress.fromNumber(111);
+  const receiver = AztecAddress.fromNumber(222);
+
+  let token: ContractInstanceWithAddress;
+  let simTester: AvmSimulationTester;
+
+  beforeEach(async () => {
+    simTester = await AvmSimulationTester.create();
+    const constructorArgs = [admin, /*name=*/ 'Token', /*symbol=*/ 'TOK', /*decimals=*/ new Fr(18)];
+    token = await simTester.deriveAndAddContractClassAndInstance(
+      constructorArgs,
+      /*deployer=*/ admin,
+      TokenContractArtifact,
+    );
+
+    const constructorResult = await simTester.simulateEnqueuedCall(
+      'constructor',
+      constructorArgs,
+      token.address,
+      /*sender=*/ admin,
+      TokenContractArtifact,
+    );
+    expect(constructorResult.reverted).toBe(false);
+  });
+
+  it('token mint, transfer, burn (and check balances)', async () => {
+    const mintAmount = new Fr(100);
+    const transferAmount = new Fr(50);
+    const nonce = new Fr(0);
+
+    await checkBalance(sender, Fr.zero());
+
+    const mintResult = await simTester.simulateEnqueuedCall(
+      'mint_to_public',
+      /*args=*/ [/*to=*/ sender, mintAmount],
+      token.address,
+      /*sender=*/ admin,
+      TokenContractArtifact,
+    );
+    expect(mintResult.reverted).toBe(false);
+    await checkBalance(sender, mintAmount);
+
+    const transferResult = await simTester.simulateEnqueuedCall(
```

Comment:
(will make sense after you read other comments)
you see it might be nice here to just make calls like in the e2e tests (with aztecjs) and then just tell the tester "do it".